### DISCUSSION TITLE
How can I directly read information from a MySQL database?

### DISCUSSION THREAD

**Discussion by bbzcx** [Kanaries/pygwalker#500] [Q&A] [answered]
Directly obtaining data from the database is more convenient than CSV files. Beneficial for later data analysis. But I tried many methods, but there was no way. Maybe it's due to Pandas restrictions.

**Comment by longxiaofei** [comment, marked as answer]
1. use latest pre-release pygwalker:

    `pip install --upgrade --pre pygwalker`

2. install extra database client

    `pip install pymysql`

example:

```python
from pygwalker.data_parsers.database_parser import Connector
import pygwalker as pyg
 
conn = Connector(
    "mysql+pymysql://username:password@host/database/schema",
    """
        SELECT
            *
        FROM
            XXX
    """
)

pyg.walk(conn)
```

more detail refer it: [https://pygwalker-docs.vercel.app/api-reference/dataset](https://pygwalker-docs.vercel.app/api-reference/dataset#database)
